src/client/voice/ClientVoiceManager.js

```diff
@@ -29,6 +35,36 @@ class ClientVoiceManager {
      * @type {VoiceBroadcast[]}
      */
     this.broadcasts = [];
+
+    /**
+     * Maps guild IDs to adapters created for use by @discordjs/voice
+     * @type {Collection<Snowflake, DiscordGatewayAdapterLibraryMethods>}
+     * @private
+     */
+    this.adapters = new Collection();
+
+    client.ws.on(WSEvents.VOICE_SERVER_UPDATE, payload => {
+      this.adapters.get(payload.guild_id)?.onVoiceServerUpdate(payload);
+    });
+
+    client.ws.on(WSEvents.VOICE_STATE_UPDATE, payload => {
+      if (payload.guild_id && payload.session_id && payload.user_id === client.user?.id) {
+        this.adapters.get(payload.guild_id)?.onVoiceStateUpdate(payload);
+      }
+    });
+
+    const onShardDisconnect = id => {
+      for (const [guildID, adapter] of this.adapters.entries()) {
+        const guild = client.guilds.cache.get(guildID);
+        if (guild?.shardID === id) {
+          adapter.destroy();
+          this.adapters.delete(guildID);
+        }
+      }
+    };
+
+    client.on('shardDisconnect', (event, id) => onShardDisconnect(id));
+    client.on('shardReconnecting', id => onShardDisconnect(id));
```

Comment:
So... on reconnecting we destroy it entirely...will it reconnect / recreate after it successfully reconnects?

Reply:
That's up to the developer (if they notice that a particular shard has reconnected, they can respawn any disconnected voice connections if they want to)